### PR TITLE
Fix localtime/gmtime usage

### PR DESCRIPTION
**Pull request by pepone**
This PR replaces localtime/gmtime usage by localtime_r/gmtime_r for platforms that support it

### CODE REVIEW

File: cpp/src/IceSSL/OpenSSLCertificateI.cpp [hunk 1]
```diff
@@ -215,18 +215,22 @@ ASMUtcTimeToTime(const ASN1_UTCTIME* s)
...
-#   pragma warning(disable:4996) // localtime is depercated
-#endif
     time_t tzone;
     {
         IceUtilInternal::MutexPtrLock<IceUtil::Mutex> sync(mut);
         time_t now = time(0);
-        tzone = mktime(localtime(&now)) - mktime(gmtime(&now));
-    }
 #if defined(_MSC_VER)
+#   pragma warning(disable:4996) // localtime is depercated
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply cpp/src/IceSSL/OpenSSLCertificateI.cpp:223]: We should probably use localtime_s and gmtime_s on Windows.

[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/IceSSL/OpenSSLCertificateI.cpp]: Fixed the code to use localtime_s and gmtime_s on Windows

File: cpp/src/IceSSL/OpenSSLCertificateI.cpp [hunk 2]
```diff
@@ -215,18 +215,21 @@ ASMUtcTimeToTime(const ASN1_UTCTIME* s)
     //
     // Windows doesn't support the re-entrant _r versions.
```
Comment (bernardnormier) [inline cpp/src/IceSSL/OpenSSLCertificateI.cpp:216]: I would remove this comment.

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Pull request merged**